internal/infrastructure/kubernetes/proxy_infra.go

```diff
@@ -23,6 +23,11 @@ func (i *Infra) CreateOrUpdateProxyInfra(ctx context.Context, infra *ir.Infra) e
 		return errors.New("infra proxy ir is nil")
 	}
 
+	if len(infra.Proxy.Listeners) == 0 {
```

Comment:
what is the impact of this ? do we see an error in the runner for this `publish`, but for the next iteration once the `listeners` spec is fixed, the next publish works fine, and the infra is created ?

Reply:
exactly, so when we will process the gateway with 1 listener that fails validation, we would see
```bash
2024-03-27T17:49:41.202+0100    INFO    gateway-api     runner/runner.go:56     received an update      {"runner": "gateway-api"}
2024-03-27T17:49:41.204+0100    INFO    gateway-api     runner/runner.go:104    proxy:
  metadata:
    labels:
      gateway.envoyproxy.io/owning-gateway-name: example
      gateway.envoyproxy.io/owning-gateway-namespace: envoy-gateway-system
  name: envoy-gateway-system/example
        {"runner": "gateway-api", "infra-ir": "envoy-gateway-system/example"}
2024-03-27T17:49:41.204+0100    INFO    infrastructure  runner/runner.go:59     received an update      {"runner": "infrastructure"}
2024-03-27T17:49:41.204+0100    ERROR   infrastructure  runner/runner.go:70     failed to create new infra      {"runner": "infrastructure", "error": "infra proxy listeners ir is nil"}
2024-03-27T17:49:41.204+0100    ERROR   watchable       message/watchutil.go:56 observed an error       {"runner": "infrastructure", "error": "infra proxy listeners ir is nil"}
2024-03-27T17:49:41.205+0100    INFO    gateway-api     runner/runner.go:115    accessLog:
  text:
  - path: /dev/stdout
        {"runner": "gateway-api", "xds-ir": "envoy-gateway-system/example"}
```
In case of example mentioned in the issue, creating valid secret, reconciles it and infra is created
```bash
2024-03-28T13:42:09.579+0100    INFO    infrastructure  runner/runner.go:59     received an update      {"runner": "infrastructure"}
2024-03-28T13:42:09.579+0100    INFO    gateway-api     runner/runner.go:104    proxy:
  listeners:
  - address: null
    name: envoy-gateway-system/example/https-example
    ports:
    - containerPort: 10443
      name: https-example
      protocol: HTTPS
      servicePort: 443
  metadata:
    labels:
      gateway.envoyproxy.io/owning-gateway-name: example
      gateway.envoyproxy.io/owning-gateway-namespace: envoy-gateway-system
  name: envoy-gateway-system/example
        {"runner": "gateway-api", "infra-ir": "envoy-gateway-system/example"}
```

Reply:
I have tested it with multiple gateways with multiple listeners and also on merged gateways

Reply:
cool thanks, rethinking, this is not really a infra layer error but a user enforced error, any way we can quietly skip this in infra layer or gateway api layer, an info log helps

Reply:
not ideal, I agree, let's try to state the error more user-friendly :)